File: challenges/263/python/solution.py
#!/usr/bin/env python
from challenges.utils import expect


class Solution:
    def isUgly(self, n: int) -> bool:
        if n == 0:
            return False

        prime_factors = [2, 3, 5]
        for prime_factor in prime_factors:
            while n % prime_factor == 0:
                n /= prime_factor

        return n == 1


if __name__ == "__main__":
    obj = Solution()
    expect(obj.isUgly(5), True)
    expect(obj.isUgly(6), True)
    expect(obj.isUgly(1), True)
    expect(obj.isUgly(14), False)

File: challenges/263/js/solution.ts
import { expect } from "../../util";

const isUgly = (n: number): boolean => {
  if (n === 0) return false;

  const primeFactors = [2, 3, 5];
  for (const primeFactor of primeFactors) {
    while (n % primeFactor === 0) n /= primeFactor;
  }

  return n === 1;
};

expect(isUgly(5), true);
expect(isUgly(6), true);
expect(isUgly(1), true);
expect(isUgly(14), false);
